### ISSUE TITLE
Uncommitting moves the commit's changes to the default branch

### ISSUE THREAD

**Issue by nathanwhit** [gitbutlerapp/gitbutler#5317]
### Version

0.13.7

### Operating System

macOS

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

When uncommitting a commit, GitButler puts the changes that were in the commit into the default branch, instead of the branch the commit was in.

https://github.com/user-attachments/assets/71ac27f6-f914-481f-8669-f58e650743b1

### How to reproduce

Create two virtual branches. Make a commit on branch A. Set branch B to be the default branch. Uncommit the commit on branch A. Notice that the changes that were in the commit have been staged on branch B, instead of branch A.

### Expected behavior

Uncommitting a commit should retain the branch of the commit's contents, regardless of the default branch.

**Comment by Byron** [comment]
Thanks a lot for reporting, the video makes the issue abundantly clear as well. Besides that, I was able to reproduce it with ease, too.

I also think that the current behaviour is unintuitive and one would expect uncommitted changes to be associated with their lane.

**Comment by estib-vega** [comment]
Hey @Byron, if you're not already looking into this, I can take a look :)

**Comment by Byron** [comment]
Yes, that would be great if you did, thank you. *(I will always assign myself if I plan to pick an issue up)*

**Comment by estib-vega** [comment]
The fix should be available in the next release :)
